docs/jep/11859-type-annotations.md

```diff
@@ -0,0 +1,160 @@
+# Type Annotation Roadmap for JAXS
+_jakevdp@_
+_Aug 2022_
+
+## Background
+Python 3.0 introduced optional function annotations ([PEP 3107](https://peps.python.org/pep-3107/)), which were later codified for use in static type checking around the release of Python 3.5 ([PEP 484](https://peps.python.org/pep-0484/)).
+To some degree, type annotations and static type checking have become an integral part of many Python development workflows, and to this end we have added annotations in a number of places throughout the JAX API.
+The current state of type annotations in JAX is a bit patchwork, and efforts to add more have been hampered by more fundamental design questions.
+This doc attempts to summarize those issues and generate a roadmap for the goals and non-goals of type annotations in JAX.
+
+Why do we need such a roadmap? Better/more comprehensive type annotations are a frequent request from users, both internally and externally.
+In addition, we frequently receive pull requests from external users seeking to improve JAX's type annotations: it's not always clear to the JAX team member reviewing the code whether such contributions are beneficial, particularly when they introduce complex Protocols to address the challenges inherent to full-fledged annotation of JAX's use of Python.
+This document details JAX's goals and recommendations for type annotations within the package.
+
+## Why Type Annotations?
+There are a number of reasons that a Python project might wish to annotate their code-base; I'll summarize them in this document as Level 1, Level 2, and Level 3.
```

Comment:
Nit: Maybe "I'll" -> "we'll", since across the doc you say "we" (e.g. "we may", "we could", "we have", etc), even though there's only one author.

```suggestion
There are a number of reasons that a Python project might wish to annotate their code-base. We'll summarize them in this document as Level 1, Level 2, and Level 3.
```